### PR TITLE
🐛 source-mssql: support `Read Committed` snapshot isolation level

### LINKED ISSUE #28543: 🐛 source-mssql: connector forces snapshot isolation level
### Connector Name

source-mssql

### Connector Version

1.1.0

### What step the error happened?

Configuring a new connector

### Revelant information

PR with proposed fix: https://github.com/airbytehq/airbyte/pull/28545

There appears to be a bug in source-mssql's support for CDC + Read Committed. Currently both snapshot isolation levels check the db for enabled full `Snapshot Isolation`, which it should not do when the `Read Committed` isolation level is selected.
 
Support for non-snapshot CDC is added here: https://github.com/airbytehq/airbyte/pull/13168 
In the connector spec we see a rework of `replication_method` to support the `Read Committed` isolation level.  `replication_method` field is renamed to `replication` for hygiene.
We can see from comments that the new implementation since 0.4.0 should use the `replication` parameter instead of the `replication_method`. More recently however, the connector was modified in https://github.com/airbytehq/airbyte/pull/16215 to use `replication_method` which has caused some problems.

In the original change to support `Read Committed` snapshot isolation level for MSSQL + CDC,  we see the new `REPLICATION_FIELD` value pulled into `CdcMssqlHelper.java`:
https://github.com/airbytehq/airbyte/blob/4284ce6b17d324887f3e51226ee481484854c7fd/airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcHelper.java#L15-L20

`isCdc` checks whether CDC is enabled. It supports both `LEGACY_REPLICATION_FIELD` and `REPLICATION_FIELD`:
https://github.com/airbytehq/airbyte/blob/4284ce6b17d324887f3e51226ee481484854c7fd/airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcHelper.java#L95-L107

`getSnapshotIsolationConfig` checks the snapshot isolation level:

https://github.com/airbytehq/airbyte/blob/4284ce6b17d324887f3e51226ee481484854c7fd/airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcHelper.java#L109-L122

Since then, the spec parameter was modified from `replication` back to `replication_method`.  This means that the `getSnapshotIsolationConfig` function still needs `REPLICATION_FIELD` to be `hasNonNull`. Unfortunately however it's always null, because `REPLICATION_FIELD` doesn't exist anymore, we only have `LEGACY_REPLICATION_FIELD`.

So now, if we look at master branch, we observe that the connector config is not passed into some functions.
i.e.,

https://github.com/airbytehq/airbyte/blob/7e4797d90dc7ddfdef2bc1ba85d6a91550397da2/airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcHelper.java#L120-L140

tl;dr quick/non-breaking-change fix is to update functions with the correct config parameter

Since this work was originally done to make naming consistent throughout CDC connectors (a reasonable goal) we should probably look at refactoring.

### Relevant log output

_No response_

### Contribute

- [X] Yes, I want to contribute

### PR DESCRIPTION
**Pull request by sh4sh**
## What
Fixes https://github.com/airbytehq/airbyte/issues/28543

## How
This PR modifies two functions `getSnapshotIsolationConfig`, `getDataToSyncConfig` and their applicable tests so that they are able to access the current connector config values.

## Recommended reading order
1. `airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcHelper.java`
2. `airbyte-integrations/connectors/source-mssql/src/test/java/io/airbyte/integrations/source/mssql/MssqlCdcHelperTest.java`

## 🚨 User Impact 🚨
This is a patch fix and should not cause breaking changes.

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcHelper.java [hunk 1]
```diff
@@ -120,8 +120,8 @@ static boolean isCdc(final JsonNode config) {
   @VisibleForTesting
   static SnapshotIsolation getSnapshotIsolationConfig(final JsonNode config) {
     // new replication method config since version 0.4.0
-    if (config.hasNonNull(REPLICATION_FIELD)) {
-      final JsonNode replicationConfig = config.get(REPLICATION_FIELD);
+    if (config.hasNonNull(LEGACY_REPLICATION_FIELD) && config.get(LEGACY_REPLICATION_FIELD).isObject()) {
```
Comment (sh4sh) [inline airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcHelper.java:123]: Maybe I should also add `isText` case here too?

**Comment by sh4sh** [comment]
todo: bump version and add changelog

**Comment by octavia-squidington-iii** [comment]
##  source-mssql-strict-encrypt test report (commit [`9c7d5a46cb`](https://github.com/airbytehq/airbyte/pull/28545/commits/9c7d5a46cb8f9d4c3fc14f22d3be907ee55270bd)) - ✅

⏲️  Total pipeline duration: 16mn05s 

| Step   | Result   |
|--------|----------|

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/sh4sh_mssql-snapshot-isolation/1689888196/9c7d5a46cb8f9d4c3fc14f22d3be907ee55270bd/source-mssql-strict-encrypt/1.1.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mssql-strict-encrypt test
```

**Comment by octavia-squidington-iii** [comment]
##  source-mssql test report (commit [`9c7d5a46cb`](https://github.com/airbytehq/airbyte/pull/28545/commits/9c7d5a46cb8f9d4c3fc14f22d3be907ee55270bd)) - ✅

⏲️  Total pipeline duration: 19mn22s 

| Step   | Result   |
|--------|----------|

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/sh4sh_mssql-snapshot-isolation/1689888196/9c7d5a46cb8f9d4c3fc14f22d3be907ee55270bd/source-mssql/1.1.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mssql test
```

File: airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcHelper.java (hunk 1 above)
Comment (rodireich) [reply airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcHelper.java:123]: `.isTextual()` is for when replication_method [was a string](https://github.com/airbytehq/airbyte/commit/90b757b22909a925a04ce172812227ed71f6f27d#diff-09d06a92478799fe02750607b91c60454858f7829e4338e61cc5071fd90d3cf4)
`.isObject()` I think is suitable for a `oneOf` selection field.
Current implementation is with `oneOf` so it sounds like this is correct (?)

**Comment by rodireich** [comment]
@sh4sh This Looks good to me.
I just wonder if the pre-0.4.0 support is still something we need to keep.
Together with the back and forth on "replication_method" to "replication" and back, This is the most confusing piece of code

**Review by rodireich — APPROVED** [review]

**Comment by marcosmarxm** [comment]
> 

I said to Sunny probably we can work in a next PR to remove all confusing code between legacy and the other one.

**Comment by octavia-squidington-iii** [comment]
##  source-mssql test report (commit [`1b88d0ec43`](https://github.com/airbytehq/airbyte/pull/28545/commits/1b88d0ec430691529c2cf12731946a075487ef8d)) - ✅

⏲️  Total pipeline duration: 18mn01s 

| Step   | Result   |
|--------|----------|

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/sh4sh_mssql-snapshot-isolation/1690214096/1b88d0ec430691529c2cf12731946a075487ef8d/source-mssql/1.1.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mssql test
```

**Comment by octavia-squidington-iii** [comment]
##  source-mssql-strict-encrypt test report (commit [`1b88d0ec43`](https://github.com/airbytehq/airbyte/pull/28545/commits/1b88d0ec430691529c2cf12731946a075487ef8d)) - ❌

⏲️  Total pipeline duration: 15mn28s 

| Step                                                                                   | Result   |
|:---------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-mssql-strict-encrypt/metadata.yaml     | ✅        |
| Connector version semver check                                                         | ✅        |
| QA checks                                                                              | ✅        |
| Build connector tar                                                                    | ✅        |
| Build source-mssql-strict-encrypt docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:source-mssql-strict-encrypt:integrationTest | ❌        |
| Acceptance tests                                                                       | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/sh4sh_mssql-snapshot-isolation/1690214096/1b88d0ec430691529c2cf12731946a075487ef8d/source-mssql-strict-encrypt/1.1.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mssql-strict-encrypt test
```

**Comment by octavia-squidington-iii** [comment]
##  source-mssql-strict-encrypt test report (commit [`4e9738166e`](https://github.com/airbytehq/airbyte/pull/28545/commits/4e9738166e9ae6b31c1fdb8a75a25abe38b0e8b6)) - ❌

⏲️  Total pipeline duration: 12mn14s 

| Step                                                                                   | Result   |
|:---------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-mssql-strict-encrypt/metadata.yaml     | ✅        |
| Connector version semver check                                                         | ✅        |
| QA checks                                                                              | ❌        |
| Build connector tar                                                                    | ✅        |
| Build source-mssql-strict-encrypt docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:source-mssql-strict-encrypt:integrationTest | ❌        |
| Acceptance tests                                                                       | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci-internal/connectors/test/pull_request/sh4sh_mssql-snapshot-isolation/1690919430/4e9738166e9ae6b31c1fdb8a75a25abe38b0e8b6/source-mssql-strict-encrypt/1.1.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mssql-strict-encrypt test
```

**Comment by octavia-approvington** [comment]
It's time <br/> fine lets go

**Review by octavia-approvington — APPROVED** [review]

**Pull request merged**